server/src/services/user.service.ts

```diff
@@ -60,8 +60,11 @@ export class UserService {
     return this.findOrFail(auth.user.id, {}).then(mapUser);
   }
 
-  create(createUserDto: CreateUserDto): Promise<UserResponseDto> {
-    return this.userCore.createUser(createUserDto).then(mapUser);
+  async create(dto: CreateUserDto): Promise<UserResponseDto> {
+    const user = await this.userCore.createUser(dto);
+    const tempPassword = user.shouldChangePassword ? dto.password : undefined;
```

Comment:
Should this logic exist here, feels specific to the email and not the user service? I'm not a big fan of sending passwords in the emails, but seems fine if it's temporary. Perhaps this should happen within the job or the template though and just pass the shouldChangePassword field across to the job/template.

Reply:
The only problem is the new password is only available in the original request that is sent to the user service (after that it is hashed). I'm fine not sending it, but then they also don't know how to actually login.

Reply:
I've seen many cases you get sent a random password for initial logged in anyway, if the user get sent an email with a default password, the user should be required to change password on first logged in

Reply:
Currently the password is only sent if the "shouldChangePassword" value is set to true.

Reply:
IMO a good approach is to send an authorized link straight to the change password page. But I'm fine leaving that for later.

Reply:
Yea, once we have reset password that may be a good way to do it. For now my suggestion was really should that line of code exist in the user service or should it exist in the template sending code. I'm not that bothered either way.